plugins/node/opentelemetry-instrumentation-cucumber/src/instrumentation.ts

```diff
@@ -0,0 +1,347 @@
+          [...hooks, ...steps].forEach(method => {
+            this._unwrap(moduleExports, method);
+          });
+        },
+        [
+          new InstrumentationNodeModuleFile<{
+            default: { new (): TestCaseRunner; prototype: TestCaseRunner };
+          }>(
+            '@cucumber/cucumber/lib/runtime/test_case_runner.js',
+            ['*'],
+            (moduleExports, moduleVersion) => {
+              diag.debug(
+                `Applying patch for @cucumber/cucumber/lib/runtime/test_case_runner.js@${moduleVersion}`
+              );
+              if (isWrapped(moduleExports.default.prototype.run)) {
+                this._unwrap(moduleExports.default.prototype, 'run');
+                this._unwrap(moduleExports.default.prototype, 'runStep');
+              }
+              this._wrap(
+                moduleExports.default.prototype,
+                'run',
+                this._getTestCaseRunPatch()
+              );
+              this._wrap(
+                moduleExports.default.prototype,
+                'runStep',
+                this._getTestCaseRunStepPatch()
+              );
+              return moduleExports;
+            },
+            (moduleExports, moduleVersion) => {
+              if (moduleExports === undefined) return;
+              diag.debug(
+                `Removing patch for @cucumber/cucumber/lib/runtime/test_case_runner.js@${moduleVersion}`
+              );
+              this._unwrap(moduleExports.default.prototype, 'run');
+              this._unwrap(moduleExports.default.prototype, 'runStep');
+            }
+          ),
+        ]
+      ),
+    ];
+  }
+
+  private static mapTags(tags: readonly messages.Tag[]): string[] {
+    return tags.map(tag => tag.name);
+  }
+
+  private static setSpanToError(span: Span, error: any) {
+    span.recordException(error);
+    span.setStatus({
+      code: SpanStatusCode.ERROR,
+      message: error?.message ?? error,
+    });
+  }
+
+  private static setSpanToStepStatus(
+    span: Span,
+    status: messages.TestStepResultStatus,
+    context?: string
+  ) {
+    if (
+      [
+        messages.TestStepResultStatus.UNDEFINED,
+        messages.TestStepResultStatus.AMBIGUOUS,
+        messages.TestStepResultStatus.FAILED,
+      ].includes(status)
+    ) {
+      span.recordException(new Error(status));
+      span.setStatus({
+        code: SpanStatusCode.ERROR,
+        message: context || status,
+      });
+    }
+    if (
+      [
+        messages.TestStepResultStatus.SKIPPED,
+        messages.TestStepResultStatus.PENDING,
+      ].includes(status)
+    ) {
+      span.addEvent(status);
+    }
+  }
+
+  private _getTestCaseRunPatch() {
+    const instrumentation = this;
+    return function (original: TestCaseRunner['run']): TestCaseRunner['run'] {
+      return async function (this: TestCaseRunner, ...args) {
+        if (!instrumentation.isEnabled()) return original.apply(this, args);
```

Comment:
https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1252/commits/2feffec0a210b5264dc96c6d941cc12a34ebc6d6
These two could be safely removed. The other two patch the callbacks that users of cucumber pass as step definitions, cucumber registers stores these functions internally, so we can't unpatch them. The check there is still needed.

Reply:
Cool, thanks for taking a look.
If you want, it could be great to include a comment above these lines with your findings for the next person to read this code :)